Src/firmware/mainf.cpp

```diff
@@ -345,55 +388,82 @@ void setup() {
 
   // Initialize Diff Drive Controller
   dc.init(params);
+
+  status = AgentStatus::CONNECTING_TO_AGENT;
+  battery_led_status = BatteryLedStatus::NOT_CONNECTED;
 }
 
 void loop() {
-  // Try to connect to uros agent
-  if (!ros_initialized && rmw_uros_ping_agent(1000, 1) == RMW_RET_OK) {
-    ros_initialized = initROS();
-    if (ros_initialized) {
-      uros_agent_connected = true;
-      (void)!rcl_timer_call(&sync_timer);
-    } else
+  switch (status) {
+    case AgentStatus::CONNECTING_TO_AGENT:
+      // Try to connect to uros agent
+      if (rmw_uros_ping_agent(1000, 1) == RMW_RET_OK) {
+        if (initROS()) {
+          (void)!rcl_timer_call(&sync_timer);
+          boot_enter_time = time();
+          status = AgentStatus::BOOT;
+        } else
+          finiROS();
+      }
+      break;
+    case AgentStatus::BOOT:
+      rclc_executor_spin_some(&executor, 0);
+
+      if (reload_parameters.exchange(false)) {
+        params.update(&param_server);
+        dc.updateParams(params);
+      }
+
+      if (publish_param_trigger) {
+        (void)!rcl_publish(&param_trigger_pub, &param_trigger, NULL);
+        publish_param_trigger = false;
+      } else if (boot_request ||
+                 time() - boot_enter_time >= BOOT_TIMEOUT) {
+        uros_agent_connected = true;
```

Comment:
We can get rid of `uros_agent_connected` variable and instead only use the `status` enum, even in ping timer callback

Reply:
Done